User_Story#1/SortingApp/BusinessLayer/CollectionSortHandler.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System;
+using System.IO;
+using BusinessLayer.Reader;
+using BusinessLayer.SorterFactory;
+using BusinessLayer.SortingAlgorithms;
+using BusinessLayer.Utilities.Parser.Interfaces;
+using BusinessLayer.Writer;
+using Helper;
+
+namespace BusinessLayer
+{
+    /// <summary>
+    /// Handles all requests regarding reading, sorting collection and writing it to the file.
+    /// </summary>
+    /// <owner>Anton Petrenko</owner>
+    /// <typeparam name="T">Certain input type.</typeparam>
+    public class CollectionSortHandler<T> where T : IComparable
+    {
+        /// <summary>
+        /// Holds the parser.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        private readonly IDataParser<T> parser;
+
+        /// <summary>
+        /// Holds the reader.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        private readonly DataReader reader;
+
+        /// <summary>
+        /// Holds the sorter.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        private CollectionSorterBase<T> sorter;
+
+        /// <summary>
+        /// Holds the writer.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        private readonly DataWriter<T> writer;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CollectionSortHandler{T}"/> class.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        /// <param name="reader">The reader instance.</param>
+        /// <param name="writer">The writer instance.</param>
+        /// <param name="sorter">The sorter instance.</param>
+        /// <param name="parser">The parser instance.</param>
+        public CollectionSortHandler(DataReader reader, DataWriter<T> writer, IDataParser<T> parser)
+        {
+            if (reader != null && writer != null && parser != null)
+            {
```

Comment:
This condition does not matches all existing variants.
please wright in this way:
if(reader is null)
   throw new ArgumentNullException(nameof(reader));

if(parser is null)
   throw new ArgumentNullException(nameof(parser));

if(writer is null)
   throw new ArgumentNullException(nameof(writer));